File: DSAlgorithms/Leetcode/ts/src/2_pointers/1984-min_dif.ts
const minimumDifference  = (nums: number[], k: number): number  => {
    nums.sort((a, b) => a - b);
    console.log(nums);
    let left = 0;
    let right = k - 1;
    let result = Infinity;
    while(right < nums.length)
    {
        result = Math.min(result, (nums[right] - nums[left]));
        left++;
        right++;
    }
    return result;
};

File: dsa_algo/leetcode/2_pointers/assorted/c++/1984-min_dif.cpp
class Solution {
public:
    int minimumDifference(vector<int>& nums, int k) {
       std::sort(nums.begin(), nums.end());
       int left = 0;
       int right = k - 1;
        //requires #include <limits>
        int result =  std::numeric_limits<int>::max();
        while(right < nums.size())
        {
            result = std::min(result, nums[right] - nums[left]);
            left ++;
            right++;
        }
        return result;
    }
};